app/controllers/concerns/api_authenticated.rb

```diff
@@ -7,15 +7,20 @@ module ApiAuthenticated
     before_action :authenticate_user!
   end
 
+  def authenticated?
+    api_admin_token? || prx_auth_token
+  end
+
   def authenticate_user!
-    user_not_authorized unless prx_auth_token
+    user_not_authorized unless authenticated?
   end
 
-  def cache_show?
-    false
+  # don't bother calculating cache keys if user will be 401'd anyways
+  def index_cache_path
+    super if authenticated?
```

Comment:
Here was another gotcha - some part of `hal_api-rails` was checking the `index_cache_path` _before_ we had determined if you're logged in or not.  This was firing off some SQL to get the items in the collection, which failed because `authorization` is null.